test/test_linalg.py

```diff
@@ -5782,6 +5782,25 @@ def test_matmul_out_kernel_errors_with_autograd(self, device, dtype):
         with torch.no_grad():
             torch.matmul(a, b, out=c)
 
+    @dtypes(torch.float, torch.complex64)
+    def test_tensordot_out_kernel_errors_with_autograd(self, device, dtype):
+        a = torch.empty((4, 2), device=device, dtype=dtype, requires_grad=True)
+        b = torch.empty((2, 4), device=device, dtype=dtype, requires_grad=True)
+        c = torch.empty((2, 2), device=device, dtype=dtype, requires_grad=True)
+        d = torch.empty((4, 4), device=device, dtype=dtype, requires_grad=False)
+        err_msg = "the 'out' tensor was specified and requires gradients"
+
+        with torch.set_grad_enabled(True), self.assertRaisesRegex(RuntimeError, err_msg):
+            torch.tensordot(a, b, dims=([1], [0]), out=c)
+
+        with torch.set_grad_enabled(True):
+            torch.tensordot(a, b, dims=([1], [0]), out=d)
+
+        with torch.set_grad_enabled(False), warnings.catch_warnings(record=True) as w:
+            c.requires_grad = False
```

Comment:
Why is this necessary when grad mode is already False?

Reply:
> Why is this necessary when grad mode is already False?
I see this for CUDA tensor it is going into this route https://github.com/pytorch/pytorch/blob/67d3053d4b25c7d2f9be33c2fbe18f1e68fdd5ef/torch/csrc/autograd/VariableTypeManual.cpp#L248C1-L249C1 unlike CPU tensor. 

This hack is just to have CUDA tensor passing the test.

Reply:
Ah interesting, thanks for investigating. Maybe worth adding a small comment about this here.

Reply:
For sure, thanks!